### ISSUE TITLE
Updated to 106.6.2 Unable to Serialize

### ISSUE THREAD

**Issue by Varorbc** [restsharp/RestSharp#1242]
use restsharp 106.5.4 is ok,but updated to 106.6.2 unable to serialize

## Expected Behavior
request.Parameters exist content

## Actual Behavior
request.Parameters.Count is 0

## Steps to Reproduce the Problem

  1.use restsharp 106.6.2
  2.    var client = new RestClient(baseUrl);
    var request = new RestRequest(Method.POST)
    {
    	XmlSerializer = new XmlSerializer()
    };
    request.AddXmlBody(data);
var response = client.Execute(request);

## Specifications

  - Version:106.6.2
  - Platform:win10 net472
  - Subsystem:net472

## StackTrace

**Comment by alexeyzimarev** [comment]
@Varorbc please don't ignore the contribution guidelines and include the information that we require for bug reports. Respect other people that work on this project and spend their time supporting open-source software.

**Comment by Varorbc** [comment]
@alexeyzimarev sorry, forgive my English is too poor

**Comment by alexeyzimarev** [comment]
@Varorbc nobody will judge your English, we need to know as many details about the issue as you can provide.

`AddBody` method does not add a conventional parameter now, it creates a new `BodyParameter`. Serialization happens when the request is executed. It is mentioned in the readme.

Unless you can't get your request executed and get a response, I don't see any issue here.

**Comment by Varorbc** [comment]
@alexeyzimarev I think you should consider the principle of downward compatibility when upgrading  RestSharp, rather than letting users make normally used programs unusable when the update is complete.

**Comment by alexeyzimarev** [comment]
@Varorbc there will be breaking changes, small or large. We cannot move on if we keep things as they are.

**Comment by alexeyzimarev** [comment]
It is fixed in 106.6.3

**Comment by Brandnew1988** [comment]
@alexeyzimarev Got the same problem after I upgraded from 106.5.4 to 106.6.5. 

However I found a solution

  RestClient client = new RestClient(ConfigurationManager.AppSettings["Url"]);
  **client.UseDotNetXmlSerializer();** (Added this after the upgrade as per the read me) 
  RestRequest request = new RestRequest(actionUrl, httpMethod);
 **request.XmlSerializer = new RestSharp.Serializers.DotNetXmlSerializer();** (and removed this after 
  the upgrade) 
  
  request.AddXmlBody(sentRequest);

  IRestResponse response = client.Execute(request);

I looks like the request is not serialized if you set the XmlSerializer on the request (is this not supported in the newest version of Restsharp? )

Anyway hope it help other people that might have the same issue.
